packages/wrangler/src/core/register-commands.ts

```diff
@@ -181,9 +182,14 @@ function createHandler(def: CommandDefinition) {
 
 			await def.validateArgs?.(args);
 
+			const configPath =
+				args.config ||
+				// @ts-expect-error args types only includes global args?
```

Comment:
I think this needs to be agnostic of things like `script`. IMO using the `config` from defineCommand is problematic at the moment because of this very reason - we use all sorts of things to define the config path in different handlers:

- args.config
- args.script
- args.name
- process.cwd()

I would suggest that unless the config path is defined by `args.config` then we should just use the manual way to get the config already defined in the handler. Otherwise we risk putting too much magic in this `createHandler` function and making it too opinionated.

In the future, we could find a way to get handlers to register their own config path, so that it could stay agnostic.

What do you think?